miette-derive/src/diagnostic_source.rs

```diff
@@ -63,14 +63,5 @@ impl CausedBy {
             },
         )
     }
-
-    pub(crate) fn gen_struct(&self) -> Option<TokenStream> {
-        let rel = &self.0;
-        Some(quote! {
-            fn diagnostic_source<'a>(&'a self) -> std::option::Option<&'a dyn miette::Diagnostic> {
-                std::option::Option::Some(self.#rel.as_ref())
-            }
-        })
-    }
```

Comment:
uhhh, this shouldn't be unused.

The `gen_struct` typically generates definitions for when the macro is used on a struct-style error, for example:

```rust
#[derive(Diagnostic, Error)]
#[error("meh")]
struct MyError {
  #[diagnostic_source]
  src: SomeError
}
```